src/test/java/dev/openfeature/sdk/providers/memory/InMemoryProviderTest.java

```diff
@@ -105,4 +114,42 @@ void shouldThrowIfNotInitialized() {
         // ErrorCode.PROVIDER_NOT_READY should be returned when evaluated via the client
         assertThrows(ProviderNotReadyError.class, ()-> inMemoryProvider.getBooleanEvaluation("fail_not_initialized", false, new ImmutableContext()));
     }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    void emitChangedFlagsOnlyIfThereAreChangedFlags() {
+        Consumer<EventDetails> handler = mock(Consumer.class);
+        Map<String, Flag<?>> flags = buildFlags();
+
+        OpenFeatureAPI.getInstance().onProviderConfigurationChanged(handler);
+        OpenFeatureAPI.getInstance().setProviderAndWait(provider);
+
+        provider.updateFlags(flags);
+
+        verify(handler, times(1))
```

Comment:
Oh, I think this test is a bit flaky. You probably will need to use Awaitility to make this assertion within some time frame, since it happens asynchronously on another thread. If you search for `await` in this repo you'll see a bunch of similar examples.

Reply:
I did this [here](https://github.com/open-feature/java-sdk/pull/1057/commits/e139de4f5653f1db5dbea56eaa5f333ec9c63d31) since it was a quick change. I hope you don't mind @ryanprayogo

Reply:
@toddbaert I do not mind, thank you.

Would it be worth adding an `.atMost(...)` so the test doesn't hang forever in the case that the assertion doesn't pass?

Reply:
There's a default time limit (I think 5s?). After that point the test fails with a timeout.

Reply:
Ah yes, you're right 👌